src/fusion_power_plant.cc

```diff
@@ -10,14 +10,98 @@ std::string FusionPowerPlant::str() {
   return Facility::str();
 }
 
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void FusionPowerPlant::EnterNotify() {
+  //Pseudocode implementation of EnterNotify()
+  cyclus::Facility::EnterNotify();
+  fuel_usage_mass = (burn_rate * (fusion_power / MW_to_GW) / seconds_per_year * context()->dt());
+  fuel_usage_atoms = fuel_usage_mass / tritium_atomic_mass;
+  blanket_turnover = blanket_size * blanket_turnover_rate;
+
+  //Truncated for readability. It goes on Tritium Storage and gets started, though.
+  fuel_startup_policy.Init(&tritium_storage).Set(fuel_incommod).Start();
+  
+  //Truncated for readability
+  blanket_fill_policy.Init(&blanket).Set(blanket_incommod).Start();
+
+  //Tritium Buy Policy Section:
+  if (refuel_mode == "schedule") {
+    //Set up a schedule fill policy with Katie's active/dormant policy code
+    //(not shown here). Do not start the policy yet.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else if (refuel_mode == "fill") {
+    //otherwise we just do it normally, but still don't start it.
+    fuel_refill_policy.Init(&tritium_storage).Set(fuel_incommod);
+  } else {
+    //This can be more sophisticated later.
+    throw cyclus::KeyError("Refill mode not recognized!");
+  }
+
+  tritium_sell_policy.Init(&tritium_excess).Set(fuel_incommod).Start();
+  helium_sell_policy.Init(&helium_storage).Set(he3).Start();
+
+}
+
 // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 void FusionPowerPlant::Tick() {
-  std::cout<<"Hello"<<std::endl;
+  //pseudocode implementation of Tick():
+  
+  if (can_operate) {
+    SequesterTritium();
+    OperateReactor();
+  }
+
+  //I think somehow combining all these would be nice...
+  DecayInventory(tritium_storage);
+  ExtractHelium(tritium_storage);
+
+  DecayInventory(tritium_excess);
+  ExtractHelium(tritium_excess);
+
+  sequestered_tritium->Decay(context()->time());
+
+  //This was a bunch of free-floating code earlier, but i've truncated it 
+  //because it's mostly for DRE stuff.
+  MoveExcessTritiumToSellBuffer();
+
+  //This maybe belongs in its own function?
+  if (context()->time() % blanket_turnover_frequency == 0 && !blanket.empty()) {
```

Comment:
This condition could be it's own function:
```suggestion
  if (blanket_cycle_time()) {
```

Reply:
Genuine question: since it's all on one line anyway, does leaving the condition written out like this not improve readability? I agree we could make it a function, but isn't that just making someone go and look at what that one-liner is somewhere else?. Maybe I'm thinking about that wrong, though.

Reply:
While your approach may be more transparent, it increases the cognitive burden.  By making it a function, the reader understands the semantic meaning of the conditions.  If they don't immediately care how that condition is met, they can just read on and understand what's happening.  If they do care how it's met, they can go and look at that function, process what it is doing in isolation, and then set aside the details of how that condition is being met when they read the overall algorithm.

Reply:
Ahh, gotcha. Makes sense. Thanks!